lib/ProxySQL_Statistics.cpp

```diff
@@ -1068,3 +1079,103 @@ void ProxySQL_Statistics::MySQL_Query_Cache_sets(SQLite3_result *resultset) {
 		statsdb_disk->execute(buf);
 	}
 }
+
+bool ProxySQL_Statistics::knows_variable_name(std::string variable_name) const
+{
+	return (variable_name_id_map.find(variable_name) != variable_name_id_map.end());
+}
+
+int64_t ProxySQL_Statistics::get_variable_id_for_name(std::string variable_name) {
+	lock_guard<mutex> lock(mu);
+
+	int64_t variable_id = -1; // Negative value indicates not yet found.
+
+	if (knows_variable_name(variable_name)) {
+		variable_id = variable_name_id_map[variable_name];
+	} else {
+		// No matching variable_id found in map. Try loading from the SQLite lookup table on disk 
+		SQLite3_result *result = NULL;
+		int cols;
+		int affected_rows;
+		char *error = NULL;
+		
+		auto select_var_id = [&]() -> int64_t {
```

Comment:
My reasoning was following along how similar code was written elsewhere in the module where these vars were shared by multiple selects. If done in similar style, this would produce near identical code but less redundant. It does also use statsdb_disk, which without reference capture would change the signature from [&]() -> int64_t to [](SQLite3DB *, const string& variable_name) -> int64_t. This would not be a simpler function signature. I'm not sure I follow on your readability argument, as the capture by reference can be treated as would any other scope in the same function.

Reply:
Yes, I see you point. I'm into everything that removes code duplication :100: . My suggestion was simply that most of the context the lambda is using, could be within the lambda scope, for me, the least shared state the lambda uses, the more simple to understand it's and it's simpler also to convert into a proper function in the future if you want. For example, promoting the `(SQLite3DB *, const string& variable_name) -> int64_t` function into a regular one, will be a matter of just copy pasting it outside de current function. So it's a matter of reuse. As you said, it can be treated as any other scope, but then, why having things in a upper scope that you can have in a lower one? Anyway, it's just a "protectionist way" of seeing it, you can ignore this as it was just a suggestion :)